conans/test/functional/old/paths_test.py

```diff
@@ -85,3 +85,64 @@ def test_with_env_variable(self, _, short_paths):
 
             self.assertEqual(self.home_short in r, short_paths)
             self.assertEqual(self.home in r, not short_paths)
+
+    def test_path_is_different_for_different_files(self, _):
+        p1 = tempfile.mkdtemp(dir=self.home)
+        short1 = path_shortener(p1, True)
+        p2 = tempfile.mkdtemp(dir=self.home)
+        short2 = path_shortener(p2, True)
+
+        self.assertNotEqual(short1, short2)
+
+    def test_path_is_deterministic(self, _):
+        p = tempfile.mkdtemp(dir=self.home)
+        short1 = path_shortener(p, True)
+        short2 = path_shortener(p, True)
+
+        self.assertEqual(short1, short2)
+
+    def test_path_is_relative_to_home_short(self, _):
+        with environment_append({"CONAN_USER_HOME_SHORT": self.home_short}):
+            p = tempfile.mkdtemp(dir=self.home)
+            r = path_shortener(p, True)
+
+        self.assertTrue(self.home_short in r)
+
+@mock.patch("subprocess.check_output", return_value=None)
+class PathShortenerWithTwoHomes(unittest.TestCase):
+    def run(self, *args, **kwargs):
+        self.home1 = tempfile.mkdtemp(suffix="_home")
+        self.home2 = tempfile.mkdtemp(suffix="_home")
+        self.home_short1 = tempfile.mkdtemp(suffix="_c")
+        self.home_short2 = tempfile.mkdtemp(suffix="_c")
+        self.file1 = os.path.join(self.home1, "testFile")
+        self.file2 = os.path.join(self.home2, "testFile")
+
+        try:
+            with environment_append({"USERNAME": "jgsogo"}):
+                super(PathShortenerWithTwoHomes, self).run(*args, **kwargs)
+        finally:
+            for f in (self.home1, self.home2, self.home_short1, self.home_short2):
+                shutil.rmtree(f, ignore_errors=True)
+
+    def test_path_is_not_the_same_for_different_homes(self, _):
```

Comment:
@memsharded This test and the conditional in https://github.com/conan-io/conan/pull/5344/files#diff-e902a4523faa9c25ee4c7e4f274eefeaR78 are to address your concern in https://github.com/conan-io/conan/issues/5284#issuecomment-498459502 regarding invalidating existing caches.

Reply:
The implementation would be less hackish without it at the cost of generating new cache entries. Would that really be a problem? The code recovers previously generated paths through the use of the link file https://github.com/conan-io/conan/pull/5344/files#diff-e902a4523faa9c25ee4c7e4f274eefeaL52.